### PR TITLE
additional tests for Slice Dictionary and Sequence types,

### LINKED ISSUE #3815: Need more test for dictionaries with optional values
Seems the only test we have for a dictionary with optional values is this

```
./IceRpc.Slice.Tests/StructTests.Slice1.slice:    i: Dictionary<int32, PingableProxy?>
```

We have more tests for sequences with optional elements but are missing a test for a sequence with optional struct elements.

### LINKED ISSUE #3816: Inconsistent naming for test enums
In enums tests we have

```
enum MyFixedLengthEnum : int16 {
    SEnum1
    SEnum2
    SEnum3
}

enum MyVarSizeEnum : varint32 {
    Enum1 = -10
    Enum2 = 20
    Enum3 = 30
}
```
What is the correct term here "length" or "size"? using both seems inconsistent

### LINKED ISSUE #3817: SliceCCSharpTask failing to load System.Text.Json 8
```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	MSB4018	The "SliceCCSharpTask" task failed unexpectedly.
System.TypeInitializationException: The type initializer for 'IceRpc.Slice.Tools.DiagnosticParser' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The system cannot find the file specified.
   at IceRpc.Slice.Tools.DiagnosticParser..cctor()
   --- End of inner exception stack trace ---
   at IceRpc.Slice.Tools.DiagnosticParser.Parse(String data)
   at IceRpc.Slice.Tools.SliceCCSharpTask.LogEventsFromTextOutput(String singleLine, MessageImportance messageImportance) in C:\Users\jose\source\repos\icerpc-csharp\tools\IceRpc.Slice.Tools\SliceCCSharpTask.cs:line 97
   at Microsoft.Build.Utilities.ToolTask.LogMessagesFromStandardErrorOrOutput(Queue dataQueue, ManualResetEvent dataAvailableSignal, MessageImportance messageImportance, StandardOutputOrErrorQueueType queueType)
   at Microsoft.Build.Utilities.ToolTask.HandleToolNotifications(Process proc)
   at Microsoft.Build.Utilities.ToolTask.ExecuteTool(String pathToTool, String responseFileCommands, String commandLineCommands)
   at Microsoft.Build.Utilities.ToolTask.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()	IceRpc.Slice.Tests	C:\Users\jose\source\repos\icerpc-csharp\tools\IceRpc.Slice.Tools\IceRpc.Slice.Tools.targets	51		
```

### PR DESCRIPTION
**Pull request by pepone**
This PR:
- adds additional tests for Slice Dictionary and Sequence types - Fixes #3815 
- renames MyFixedLengthEnum to MyFixedSizeEnum - Fixes #3816
- ~~downgrades IceRpc.Slice.Tools Json dependency - Fixes #3817~~ See #3821

### CODE REVIEW

File: tests/IceRpc.Slice.Tests/DictionaryMappingTests.slice [hunk 1]
```diff
@@ -5,9 +5,47 @@ module IceRpc::Slice::Tests
 interface DictionaryMappingOperations {
     returnDictionaryTuple() -> (r1: Dictionary<int32, int32> r2: Dictionary<int32, int32>)
 
-    returnDictionary() -> Dictionary<int32, int32>
+    returnDictionaryOfInt32() -> Dictionary<int32, int32>
+    sendDictionaryOfInt32(p: Dictionary<int32, int32>)
 
-    sendDictionary(p: Dictionary<int32, int32>)
+    returnDictionaryOfString() -> Dictionary<string, string>
```
[the remark this answers is not in the document]
Comment (pepone) [reply tests/IceRpc.Slice.Tests/DictionaryMappingTests.slice:11]: I remove redundant tests.

We have to test the different paths for https://github.com/icerpc/icerpc-csharp/blob/main/tools/slicec-cs/src/decoding.rs#L368

File: tests/IceRpc.Slice.Tests/DictionaryMappingTests.cs
Comment (externl) [inline tests/IceRpc.Slice.Tests/DictionaryMappingTests.cs:1]: Still feels like there are too many tests to me. To me we are testing only two behaviors, testing the generated request/response objects for dictionaries. 

Would it make sense to make many of these `_dictionary_of_XXX` into test cases for `Operation_returining_dictionary`? That is the unit of behavior we're testing.

Comment (pepone) [reply tests/IceRpc.Slice.Tests/DictionaryMappingTests.cs:1]: I don't follow, which ones do you think are still redundant?

Comment (externl) [reply tests/IceRpc.Slice.Tests/DictionaryMappingTests.cs:1]: I think the thing I'm trying to figure out is what our unit of behavior is with these tests. To me it seems that the behavior is to testing the generated encode/operations for dictionaries.

I don't think they're redundant, just that I think they might be better as test cases for one test.

So convert 
- Operation_returning_a_dictionary_of_string
- Operation_returning_a_dictionary_of_int32
- Operation_returning_a_dictionary_of_fixed_size_enum
- Operation_returning_a_dictionary_of_compact_struct
- Operation_returning_a_dictionary_of_optional_int32
- Operation_returning_a_dictionary_of_optional_fixed_size_enum
- Operation_returning_a_dictionary_of_optional_compact_struct
- ...

into (psudo code)
```
[TestCase(string)]
[TestCase(int)]
Operation_returning_a_dictionary() {}
```

🤷

Comment (pepone) [reply tests/IceRpc.Slice.Tests/DictionaryMappingTests.cs:1]: But here we are testing the generated code for the different types, specially the different code paths in https://github.com/icerpc/icerpc-csharp/blob/main/tools/slicec-cs/src/decoding.rs#L368

**Review by externl — APPROVED** [review]

File: tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.csproj [hunk 2]
```diff
@@ -42,7 +42,7 @@
    <ItemGroup>
       <PackageReference Include="Microsoft.Build.Framework" Version="17.8.3" />
       <PackageReference Include="Microsoft.Build.Utilities.Core" Version="17.8.3" />
-      <PackageReference Include="System.Text.Json" Version="8.0.*" />
```
Comment (bernardnormier) [inline tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.csproj:45]: Would be nicer to fix this issue in a separate PR.

**Review by bernardnormier — APPROVED** [review]

File: tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.csproj (hunk 2 above)
Comment (pepone) [reply tools/IceRpc.Slice.Tools/IceRpc.Slice.Tools.csproj]: Yes, already did

**Pull request merged**